### PR TITLE
Refactor OrganisationIndexPresenter

### PR DESCRIPTION
**Pull request by callumknights**
Change functionality of OrganisationInPresenter to more closely mirror other `publishing-api` presenters allowing the removal of `publish_organisations_index_page.rb`.

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: test/unit/app/presenters/publishing_api/organisations_index_presenter_test.rb [hunk 1]
```diff
@@ -0,0 +1,146 @@
+require "test_helper"
+
+class PublishingApi::OrganisationsPresenterTest < ActiveSupport::TestCase
+  extend Minitest::Spec::DSL
+
+  test "presents a valid content item" do
+    organisation_one = create(:organisation, organisation_type_key: :executive_office, content_id: "e08eb3d9-ade7-4a5a-9b69-35008220b77f")
+    organisation_two = create(:organisation, organisation_type_key: :ministerial_department, content_id: "0ad3f6c2-5492-4d37-86cf-27afb8d85b1c")
+    organisation_three = create(:organisation, organisation_type_key: :non_ministerial_department, content_id: "01ba16a6-2fd6-4621-9296-8afbd3114c1b")
+    organisation_four = create(:organisation, organisation_type_key: :public_corporation, content_id: "735b5c03-7937-4c68-920b-d087795fccd9")
```
Comment (brucebolt) [inline test/unit/app/presenters/publishing_api/organisations_index_presenter_test.rb:10]: Do you need to specify content IDs here?  I think the factories should generate one automatically.

**Review by brucebolt — COMMENTED** [review]
Thanks, this looks good.  I've left a few minor comments that should just make things slightly simpler.

File: test/unit/app/presenters/publishing_api/organisations_index_presenter_test.rb (hunk 1 above)
Comment (callumknights) [reply test/unit/app/presenters/publishing_api/organisations_index_presenter_test.rb:10]: This is related to the issue I discussed with you with the generated titles changing, now I've solved that I can dispense with this as well. Cheers for pointing it out.

File: lib/present_page_to_publishing_api.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ class PresentPageToPublishingApi
   def publish(presenter_class)
     payload = presenter_class.new
     Services.publishing_api.put_content(payload.content_id, payload.content)
-    Services.publishing_api.patch_links(payload.content_id, links: payload.links)
+    Services.publishing_api.patch_links(payload.content_id, links: payload.links) unless payload.is_a?(PublishingApi::OrganisationsIndexPresenter)
```
[the remark this answers is not in the document]
Comment (callumknights) [reply lib/present_page_to_publishing_api.rb]: Thanks for the idea! Both things changed!

**Review by brucebolt — APPROVED** [review]
Thanks for making those changes.  All looks good to me 👍.

**Pull request merged**